### ISSUE TITLE
winitrequires control over the principal class. You must create the event loop before other parts of your application initialize NSApplication

### ISSUE THREAD

**Issue by dezbracaty** [rust-windowing/winit#3772]
## Background 
I built a program combine Qt/Qml and bevy , and I can run it normally on ubuntu system , but when run it on macos, it output a error, and I think this is a platform related problem. So I hope to know how to solve it.

## project address
https://github.com/dezbracaty/BevyQml

## Information
After running App::new() .add_plugins(DefaultPlugins) .add_systems(Startup, setup) .add_systems(Update, animate_cube) .run();
It crash and output :
2024-07-02T05:56:27.686954Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "MacOS 15.0 ", kernel: "24.0.0", cpu: "Apple M1 Pro", core_count: "10", memory: "16.0 GiB" } thread '<unnamed>' panicked at /Users/allen/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.3/src/platform_impl/macos/event_loop.rs:220:13: winitrequires control over the principal class. You must create the event loop before other parts of your application initialize NSApplication note: run withRUST_BACKTRACE=1 environment variable to display a backtrace fatal runtime error: failed to initiate panic, error 5 [1]    33997 abort      /Users/allen/Projects/RustProjects/BevyQml/build/bevyQml/example_qml_minimal

![image](https://github.com/rust-windowing/winit/assets/20027937/da5b15fe-8de7-4ad2-a815-3e9ae07c5a22)

### macOS version

```shell
Macos 15 beta2
rustc 1.79.0 (129f3b996 2024-06-10)
```

### Winit version

> cargo tree -i winit
> winit v0.30.3
> ├── accesskit_winit v0.21.1
> │   └── bevy_winit v0.14.0-dev (https://github.com/bevyengine/bevy?branch=main#f607be87)
> │       └── bevy_internal v0.14.0-dev (https://github.com/bevyengine/bevy?branch=main#f607be87)
> │           └── bevy v0.14.0-dev (https://github.com/bevyengine/bevy?branch=main#f607be87)
> │               └── qml_minimal v0.1.0 (/Users/allen/Projects/RustProjects/BevyQml/bevyQml/rust)
> └── bevy_winit v0.14.0-dev (https://github.com/bevyengine/bevy?branch=main#f607be87) (*)

**Comment by madsmtm** [comment]
Indeed, the issue is that something is started before `EventLoop::new`, which means we no longer have control over the `NSApplication`. I plan to fix this so that Winit doesn't need control over the `NSApplication`.
